tree/ntuple/v7/test/ntuple_storage_daos.cxx

```diff
@@ -17,7 +17,7 @@ TEST(RPageStorageDaos, Basics)
 
    {
       RNTupleWriteOptions options;
-      auto ntuple = RNTupleWriter::Recreate(std::move(model), "ntuple", daosUri, options);
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "ntuple-1", daosUri, options);
```

Comment:
We should also add tests for the following cases:
- Try to create a ntuple with a name that is already used in the container (throws).
- Write two different ntuples, e.g. `ntuple-1` and `ntuple-2` -in that order-; then check that both are readable.  In particular, with this we intend to test that `ntuple-1` is still readable even after writing `ntuple-2`.
- Try to open an nonexistent ntuple in the container (throws).

Reply:
> Try to create a ntuple with a name that is already used in the container (throws).

Should this throw? Namely: should `SinkDaos::CreateImpl()` reject the operation if a given name exists? It doesn't now :)

Even if we expect a WORM use case, I assumed support for reopening, updating and / or continuing to fill an existing ntuple, since we support DAOS API update operations on existing `{d,a}keys`.

Regardless of the current decision, perhaps an `RNTupleWriteOptionsDaos::fOverwrite` option in the future?

The other tests are certainly good coverage.

Reply:
> Should this throw? Namely: should `SinkDaos::CreateImpl()` reject the operation if a given name exists? It doesn't now :)

Doesn't it throw here in that case https://github.com/root-project/root/pull/11480/files#diff-10b5750503baed69f8bd292c07f843fb8240b7ed6971761382662891987f4fc7R136?

> Even if we expect a WORM use case, I assumed support for reopening, updating and / or continuing to fill an existing ntuple, since we support DAOS API update operations on existing `{d,a}keys`.

In principle, ntuples are immutable once written - actually, appending is not supported in the file backend.

> Regardless of the current decision, perhaps an `RNTupleWriteOptionsDaos::fOverwrite` option in the future?

Yes, we might actually add that, but let's do it a follow-up PR.

Reply:
> Doesn't it throw here in that case https://github.com/root-project/root/pull/11480/files#diff-10b5750503baed69f8bd292c07f843fb8240b7ed6971761382662891987f4fc7R136?

This throws only in the event of a true hash collision; if the hash matches and the name matches, the sink is allowed to go ahead and issue update API calls. 

Given the discussion, it makes sense to (1) throw in all hash hit cases; and (2) add the appropriate DAOS flag to the update call so an existing `akey` cannot be overwritten until we support this functionality explicitly. 

Thanks !